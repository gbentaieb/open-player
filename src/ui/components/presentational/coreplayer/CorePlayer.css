.videoContainer {
  width: 100%;
  height: 100%;
}

.videoElement {
  width: 100%;
  height: 100%;
}
