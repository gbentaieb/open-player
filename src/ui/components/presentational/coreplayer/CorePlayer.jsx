/* eslint-disable no-debugger */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './CorePlayer.css';

class CorePlayer extends Component {
  static propTypes = {
    url: PropTypes.string,
    RxPlayer: PropTypes.func.isRequired,
  }

  static defaultProps = {
    url: '',
  }

  componentDidMount() {
    const { url } = this.props;

    this.setVideoElementProperties();
    this.createRxPlayer();

    if (url) {
      const options = this.getLoadVideoOptions({ url });
      this.rxPlayer.loadVideo(options);
    }
  }

  componentWillReceiveProps(nextProps) {
    const { url } = this.props;

    if (nextProps.url !== url) {
      const options = this.getLoadVideoOptions({ url: nextProps.url });
      this.rxPlayer.loadVideo(options);
    }
  }

  setVideoElementProperties() {
    this.videoElement.controls = false;
    this.videoElement.disableRemotePlayback = true;
    this.videoElement.playsinline = '';
  }

  getLoadVideoOptions = ({ url }) => (
    {
      url,
      transport: 'dash',
      autoPlay: true,
    }
  )

  createRxPlayer() {
    this.rxPlayer = new this.props.RxPlayer({
      videoElement: this.videoElement,
      limitVideoWidth: true,
      throttleWhenHidden: true,
    });

    return this.rxPlayer;
  }

  render() {
    return (
      <div>
        <video
          className={styles.video}
          ref={(v) => { this.videoElement = v; }}
        />
      </div>
    );
  }
}

export default CorePlayer;
