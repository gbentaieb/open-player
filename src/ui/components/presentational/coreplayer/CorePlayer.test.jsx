import React from 'react';
import FakeRxPlayer from '../../../../../test/fakeRxPlayer';
import CorePlayer from './CorePlayer';

test('CorePlayer > Snapshot', () => {
  const wrapper = mount(<CorePlayer RxPlayer={FakeRxPlayer} />);
  expect(wrapper).toMatchSnapshot();
});

test('CorePlayer > componentDidMount > loadVideo', () => {
  const url = 'https://www.test.com/test.mp4';
  const wrapper = mount(<CorePlayer RxPlayer={FakeRxPlayer} url={url} />);
  const instance = wrapper.instance();
  expect(instance.rxPlayer.loadVideoCalled).toEqual(true);
});

test('CorePlayer > componentDidMount > video properties', () => {
  const wrapper = mount(<CorePlayer RxPlayer={FakeRxPlayer} />);
  const instance = wrapper.instance();
  expect(instance.videoElement.controls).toEqual(false);
  expect(instance.videoElement.disableRemotePlayback).toEqual(true);
  expect(instance.videoElement.playsinline).toEqual('');
});

test('CorePlayer > componentWillReceiveProps > video url', () => {
  const url = 'https://www.test.com/test.mp4';
  const wrapper = mount(<CorePlayer RxPlayer={FakeRxPlayer} />);
  wrapper.setProps({ url });
  const instance = wrapper.instance();
  expect(instance.rxPlayer.loadVideoCalled).toEqual(true);
});

test('CorePlayer > getLoadVideoOptions', () => {
  const url = 'https://www.test.com/test.mp4';
  const result = {
    url,
    transport: 'dash',
    autoPlay: true,
  };

  const wrapper = mount(<CorePlayer RxPlayer={FakeRxPlayer} />);
  const instance = wrapper.instance();
  expect(instance.getLoadVideoOptions({ url })).toEqual(result);
});

test('CorePlayer > createRxPlayer', () => {
  const wrapper = mount(<CorePlayer RxPlayer={FakeRxPlayer} />);
  const rxPlayer = wrapper.instance().createRxPlayer();
  expect(rxPlayer.constructorCalled).toEqual(true);
});
