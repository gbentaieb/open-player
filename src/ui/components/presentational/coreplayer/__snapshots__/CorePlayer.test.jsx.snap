// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CorePlayer > Snapshot 1`] = `
<CorePlayer
  RxPlayer={[Function]}
  url=""
>
  <div>
    <video />
  </div>
</CorePlayer>
`;
